### ISSUE TITLE
Ensure algorithms return something

### ISSUE THREAD

**Issue by annevk** [immersive-web/anchors#44]
I noticed a few times that algorithms were supposed to return something, e.g., a promise, but didn't.

It might help re-reading the algorithms and see if you could write an implementation from scratch given what's written down.

**Comment by bialpio** [comment]
Ah, good catch! I'm assuming you refer to the algorithms for both variants of `createAnchor()`, I'll take a pass and make sure the algorithms should be possible to implement from scratch and that they align with prototype implementation that we have to reduce the risk of future compat issues.

**Comment by annevk** [comment]
I'm pretty sure that fix doesn't work as you sometimes terminate the set of steps early.

**Comment by bialpio** [comment]
Yes, I've noticed this and then started going through another spec to double-check - this is also how I was describing early termination in [hit test module](https://immersive-web.github.io/hit-test/). Keeping the bug open until I fix this.

**Comment by annevk** [comment]
I think I found the same issue in the main XR specification. Probably best not to copy-and-paste things too much 😉
